Throw NotSupportedException when using `SetErrorStatusOnException` in Mono Runtime and Native AOT environment. (#5374)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry/CHANGELOG.md

## Unreleased

> * Throw NotSupportedException when using `ExceptionProcessor` for Tracing in

Reviewer: `ExceptionProcessor` is an internal class which the user may not know about.

```suggestion
* Throw NotSupportedException when using `SetErrorStatusOnException` method for Tracing in
```

>   Mono Runtime and Native AOT environment because the the dependent

Reviewer: The generic `NotSupportedException` message would not mention anything about AOT.

Could you add `<remarks>` in the [code documentation ](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Trace/Builder/TracerProviderBuilderExtensions.cs#L24) for `SetErrorStatusOnException` method saying "This method is not supported for AOT as of .NET 8."?

Author: done!

>   Tracing in Mono Runtime and Native AOT environment because the the dependent

Reviewer 2: ```suggestion
  Tracing in Mono Runtime and Native AOT environment because the dependent
```

  `Marshal.GetExceptionPointers()` API is not supported on these platforms.

>   ([#5347](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5347))

Reviewer: ```suggestion
  ([#5374](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5374))
```

Reviewer 3: Good eyes 👁️ @utpilla
